nncf/pruning/filter_pruning/algo.py

```diff
@@ -280,62 +282,82 @@ def _find_layerwise_pruning_rate(self, target_flops_pruning_rate):
                 else:
-                    self._set_binary_masks_for_all_pruned_modules(pruning_rate)
-            else:
-                layerwise_pruning_rate = pruning_rate
-                if self.prune_flops:
-                    # Looking for layerwise pruning rate needed for asked flops pruning rate
-                    layerwise_pruning_rate = self._find_layerwise_pruning_rate(pruning_rate)
-                self._set_binary_masks_for_filters(layerwise_pruning_rate)
+                    # Pruning uniformly with the same pruning rate across layers
+                    layerwise_pruning_rate = pruning_rate
+                    if self.prune_flops:
+                        # Looking for layerwise pruning rate needed for the required flops pruning rate
+                        layerwise_pruning_rate = self._find_uniform_pruning_rate_for_target_flops(pruning_rate)
+                    layerwise_pruning_rate_dict = {group_id: layerwise_pruning_rate for group_id in group_ids}
+                    self._set_binary_masks_for_pruned_modules_groupwise(layerwise_pruning_rate_dict)
 
             if self.zero_grad:
                 self.zero_grads_for_pruned_modules()
+
         self._apply_masks()
         if run_batchnorm_adaptation:
             self.run_batchnorm_adaptation(self.config)
 
-    def _set_binary_masks_for_filters(self, pruning_rate):
-        nncf_logger.debug("Setting new binary masks for pruned modules.")
-
-        with torch.no_grad():
-            for group in self.pruned_module_groups_info.get_all_clusters():
-                filters_num = torch.tensor([get_filters_num(minfo.module) for minfo in group.nodes])
-                assert torch.all(filters_num == filters_num[0])
-                device = group.nodes[0].module.weight.device
-
-                cumulative_filters_importance = torch.zeros(filters_num[0]).to(device)
-                # 1. Calculate cumulative importance for all filters in group
-                for minfo in group.nodes:
-                    filters_importance = self.filter_importance(minfo.module.weight,
-                                                                minfo.module.target_weight_dim_for_compression)
-                    cumulative_filters_importance += filters_importance
-
-                # 2. Calculate threshold
-                num_of_sparse_elems = get_rounded_pruned_element_number(cumulative_filters_importance.size(0),
-                                                                        pruning_rate)
-                threshold = sorted(cumulative_filters_importance)[min(num_of_sparse_elems, filters_num[0] - 1)]
-                mask = calculate_binary_mask(cumulative_filters_importance, threshold)
-
-                # 3. Set binary masks for filter
-                for minfo in group.nodes:
-                    pruning_module = minfo.operand
-                    pruning_module.binary_filter_pruning_mask = mask
+    @property
+    def current_groupwise_pruning_rate(self):
```

Comment:
Is it used?

Reply:
this public API and is targeted for the user to obtain a dict format to use with `set_pruning_rate` (the user can simultaneously get group id's and current groupwise rates to tweak them)